packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
         r = requests.post(proxy_uri, data=cmd)
         logging.info('Requests Response: %s', repr(r.json()))
         assert r.json()['status'] == 0
+        return json.loads(r.json()['output'])
     return _ensure_routable
 
 
+def vip_test(dcos_api_session, vip, vipaddr, samehost, vipnet, proxynet):
+    teststr = 'vip={} vipaddr={} samehost={} vipnet={} proxynet={}'.format(vip, vipaddr, samehost, vipnet,
+                                                                           proxynet)
+    logging.info('VIP_TEST START {}'.format(teststr))
+    agents = list(dcos_api_session.slaves)
+    # make sure we can reproduce
+    random.seed(vip)
+    random.shuffle(agents)
+    host1 = agents[0]
+    host2 = agents[0]
+    if not samehost:
+        host2 = agents[1]
```

Comment:
This test will likely fail for any cluster that has more than 2 private agents. Either this needs to be written in a way that takes the service points from the app deployment to figure out where to ping or this whole test needs to be skipped when we have more than 2 private agents. I would vote for the former. Any reason that's not possible?